src/eko/couplings.py

```diff
@@ -737,7 +738,20 @@ def a(
         for k, seg in enumerate(path):
             # skip a very short segment, but keep the matching
             if not np.isclose(seg.origin, seg.target):
-                new_a = self.compute(final_a, seg.nf, seg.origin, seg.target)
+                nli = constants.lepton_number(seg.origin)
+                nlf = constants.lepton_number(seg.target)
+                if nli != nlf:
+                    # it means that MTAU is between origin and target:
+                    # first we evolve from origin to MTAU with nli leptons
+                    a_tmp = self.compute(
+                        final_a, seg.nf, nli, seg.origin, constants.MTAU**2
+                    )
+                    # then from MTAU to target with nlf leptons
```

Comment:
Just for curiosity: At NLO there are no matching conditions to add here ?

Reply:
that's a good question...

Reply:
I guess (!) the whole derivation goes through for any theory of a scalar boson with fermions (with the appropriate changes on couplings etc.) ... however, quoting SF from a mail on Monday "This said, it seems to me that using a VFN for leptons corresponds to using a scheme that differs from MSbar in the QED sector, which is not a very good idea!"